### DISCUSSION TITLE
Join Function

### DISCUSSION THREAD

**Discussion by fruitoiz** [tortoise/tortoise-orm#1773] [Ideas]
Can I use the join function in Tortoise?

I need to run this code on sql:
```
SELECT `t1`.id, `t1`.name, `t1`.description, `t1`.emoji, `t2`.id, `t2`.user_id, `t2`.achivka_id, `t2`.created FROM userachivki AS `t2` INNER JOIN achivki AS `t1` ON (`t1`.id = `t2`.achivka_id) WHERE (`t2`.user_id = 100) LIMIT 6 OFFSET 0
```

But I didn't find how to do it in the library.

The 'ForeignKeyField' option doesn't suit me. Tell me how to do this via 'join'. 

As far as I understand, this function does not exist. But please add it!

**Comment by henadzit** [comment]
It should be something like 

```
UserAchivka.filter(user_id=100).select_related("achivka").limit(6)
```

**Reply by fruitoiz** [reply]
This option does not suit me :(

**Reply by fruitoiz** [reply]
'join' Will it be added sometime?

**Reply by henadzit** [reply]
@fruitoiz why it does not suit you? Can you please explain your requirements and why this approach is not working you?

The tortoise query interface is modeled after Django. Can you do what you want in Django?

**Reply by fruitoiz** [reply]
> Почему это вас не устраивает? 

it is not always possible to use "ForeignKeyField" and sometimes it is not even desirable, it would be great if there was "join" like in pure SQL. This is a useful tool, I wish the wonderful TORTOISE-ORM had it too! I don't quite understand why you bypassed this functionality :(

**Reply by abondar** [reply]
Because it doesn't map with ORM nature and interface.
Implications of "just using .join()" means whole other interface needs to be created and it should work somehow. 

Like - what object these joined fields should populate? How user access these fields? If you have concrete idea how interface for that should look like and how it should be used - you can create issue describing it and make proposal of improvement

If you need just need to sometimes execute some wanky raw sql - you can use [Model.raw](https://tortoise.github.io/models.html#tortoise.models.Model.raw)
